src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -614,7 +617,24 @@ GenerateSizeQueryOnMultiplePlacements(List *shardIntervalList, char *sizeQuery)
 		char *shardQualifiedName = quote_qualified_identifier(schemaName, shardName);
 		char *quotedShardName = quote_literal_cstr(shardQualifiedName);
 
-		appendStringInfo(selectQuery, sizeQuery, quotedShardName);
+		if (sizeQueryType == RELATION_SIZE)
+		{
+			appendStringInfo(selectQuery, PG_RELATION_SIZE_FUNCTION, quotedShardName);
+		}
+		else if (sizeQueryType == TOTAL_RELATION_SIZE)
+		{
+			appendStringInfo(selectQuery, PG_TOTAL_RELATION_SIZE_FUNCTION,
+							 quotedShardName);
+		}
+		else if (sizeQueryType == CSTORE_TABLE_SIZE)
+		{
+			appendStringInfo(selectQuery, CSTORE_TABLE_SIZE_FUNCTION, quotedShardName);
+		}
+		else if (sizeQueryType == TABLE_SIZE)
+		{
+			appendStringInfo(selectQuery, PG_TABLE_SIZE_FUNCTION, quotedShardName);
+		}
+
```

Comment:
Consider a switch statement for here

Reply:
I was about to suggest a switch. The default branch could use an `elog(ERROR,....)` as it should be expected the user will never hit it.